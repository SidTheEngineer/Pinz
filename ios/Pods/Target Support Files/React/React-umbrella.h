#ifdef __OBJC__
#import <UIKit/UIKit.h>
#endif

#import "RCTAssert.h"
#import "RCTBridge+Private.h"
#import "RCTBridge.h"
#import "RCTBridgeDelegate.h"
#import "RCTBridgeMethod.h"
#import "RCTBridgeModule.h"
#import "RCTBundleURLProvider.h"
#import "RCTConvert.h"
#import "RCTDefines.h"
#import "RCTDisplayLink.h"
#import "RCTErrorCustomizer.h"
#import "RCTErrorInfo.h"
#import "RCTEventDispatcher.h"
#import "RCTFrameUpdate.h"
#import "RCTImageSource.h"
#import "RCTInvalidating.h"
#import "RCTJavaScriptExecutor.h"
#import "RCTJavaScriptLoader.h"
#import "RCTJSStackFrame.h"
#import "RCTKeyCommands.h"
#import "RCTLog.h"
#import "RCTModuleData.h"
#import "RCTModuleMethod.h"
#import "RCTMultipartDataTask.h"
#import "RCTMultipartStreamReader.h"
#import "RCTNullability.h"
#import "RCTParserUtils.h"
#import "RCTPerformanceLogger.h"
#import "RCTRootView.h"
#import "RCTRootViewDelegate.h"
#import "RCTRootViewInternal.h"
#import "RCTTouchEvent.h"
#import "RCTTouchHandler.h"
#import "RCTURLRequestDelegate.h"
#import "RCTURLRequestHandler.h"
#import "RCTUtils.h"
#import "RCTWebSocketProxy.h"
#import "RCTWebSocketProxyDelegate.h"
#import "CSSLayout.h"
#import "CSSMacros.h"
#import "CSSNodeList.h"
#import "RCTJSCErrorHandling.h"
#import "RCTJSCExecutor.h"
#import "RCTJSCWrapper.h"
#import "RCTAccessibilityManager.h"
#import "RCTAlertManager.h"
#import "RCTAppState.h"
#import "RCTAsyncLocalStorage.h"
#import "RCTClipboard.h"
#import "RCTDevLoadingView.h"
#import "RCTDevMenu.h"
#import "RCTEventEmitter.h"
#import "RCTExceptionsManager.h"
#import "RCTI18nManager.h"
#import "RCTI18nUtil.h"
#import "RCTKeyboardObserver.h"
#import "RCTRedBox.h"
#import "RCTSourceCode.h"
#import "RCTStatusBarManager.h"
#import "RCTTiming.h"
#import "RCTUIManager.h"
#import "RCTFPSGraph.h"
#import "RCTJSCProfiler.h"
#import "RCTMacros.h"
#import "RCTProfile.h"
#import "RCTActivityIndicatorView.h"
#import "RCTActivityIndicatorViewManager.h"
#import "RCTAnimationType.h"
#import "RCTAutoInsetsProtocol.h"
#import "RCTBorderDrawing.h"
#import "RCTBorderStyle.h"
#import "RCTComponent.h"
#import "RCTComponentData.h"
#import "RCTConvert+CoreLocation.h"
#import "RCTConvert+MapKit.h"
#import "RCTDatePicker.h"
#import "RCTDatePickerManager.h"
#import "RCTFont.h"
#import "RCTMap.h"
#import "RCTMapAnnotation.h"
#import "RCTMapManager.h"
#import "RCTMapOverlay.h"
#import "RCTModalHostView.h"
#import "RCTModalHostViewController.h"
#import "RCTModalHostViewManager.h"
#import "RCTNavigator.h"
#import "RCTNavigatorManager.h"
#import "RCTNavItem.h"
#import "RCTNavItemManager.h"
#import "RCTPicker.h"
#import "RCTPickerManager.h"
#import "RCTPointerEvents.h"
#import "RCTProgressViewManager.h"
#import "RCTRefreshControl.h"
#import "RCTRefreshControlManager.h"
#import "RCTRootShadowView.h"
#import "RCTScrollableProtocol.h"
#import "RCTScrollView.h"
#import "RCTScrollViewManager.h"
#import "RCTSegmentedControl.h"
#import "RCTSegmentedControlManager.h"
#import "RCTShadowView.h"
#import "RCTSlider.h"
#import "RCTSliderManager.h"
#import "RCTSwitch.h"
#import "RCTSwitchManager.h"
#import "RCTTabBar.h"
#import "RCTTabBarItem.h"
#import "RCTTabBarItemManager.h"
#import "RCTTabBarManager.h"
#import "RCTTextDecorationLineType.h"
#import "RCTView.h"
#import "RCTViewControllerProtocol.h"
#import "RCTViewManager.h"
#import "RCTWebView.h"
#import "RCTWebViewManager.h"
#import "RCTWrapperViewController.h"
#import "UIView+Private.h"
#import "UIView+React.h"
#import "RCTActionSheetManager.h"
#import "RCTLocationObserver.h"
#import "RCTGIFImageDecoder.h"
#import "RCTImageBlurUtils.h"
#import "RCTImageCache.h"
#import "RCTImageEditingManager.h"
#import "RCTImageLoader.h"
#import "RCTImageStoreManager.h"
#import "RCTImageUtils.h"
#import "RCTImageView.h"
#import "RCTImageViewManager.h"
#import "RCTLocalAssetImageLoader.h"
#import "RCTResizeMode.h"
#import "RCTLinkingManager.h"
#import "RCTDataRequestHandler.h"
#import "RCTFileRequestHandler.h"
#import "RCTHTTPRequestHandler.h"
#import "RCTNetInfo.h"
#import "RCTNetworking.h"
#import "RCTNetworkTask.h"
#import "RCTSettingsManager.h"
#import "RCTRawTextManager.h"
#import "RCTShadowRawText.h"
#import "RCTShadowText.h"
#import "RCTText.h"
#import "RCTTextField.h"
#import "RCTTextFieldManager.h"
#import "RCTTextManager.h"
#import "RCTTextSelection.h"
#import "RCTTextView.h"
#import "RCTTextViewManager.h"
#import "RCTVibration.h"
#import "RCTSRWebSocket.h"
#import "RCTWebSocketExecutor.h"
#import "RCTWebSocketManager.h"
#import "RCTWebSocketModule.h"

FOUNDATION_EXPORT double ReactVersionNumber;
FOUNDATION_EXPORT const unsigned char ReactVersionString[];

