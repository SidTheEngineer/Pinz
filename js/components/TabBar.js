import React from 'react';
import { StyleSheet } from 'react-native';
import ScrollableTabView, { DefaultTabBar } from 'react-native-scrollable-tab-view';
import Posts from '../components/Posts';
import Map from './Map';

const styles = StyleSheet.create({
  textStyle: {
    fontSize: 16,
    color: 'white'
  },

  underlineStyle: {
    backgroundColor: 'white'
  }
});

const TabBar = props => (
  <ScrollableTabView
    renderTabBar={() => <DefaultTabBar />}
    tabBarPosition="bottom"
    tabBarBackgroundColor={'#05006d'}
    tabBarTextStyle={styles.textStyle}
    tabBarUnderlineStyle={styles.underlineStyle}
  >
    <Map {...props} tabLabel="Map" />
    <Posts tabLabel="Posts" />
  </ScrollableTabView>
);

export default TabBar;
