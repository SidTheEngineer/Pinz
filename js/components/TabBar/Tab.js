import React from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

import { COLORS } from '../../constant';

const styles = StyleSheet.create({
  tab: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    width: 50,
    backgroundColor: 'rgb(245,245,245)',
    elevation: 8
  }
});

const Tab = props => (
  <TouchableOpacity
    style={styles.tab}
    onPress={() => props.goToPage(props.index)}
    activeOpacity={1}
  >
    <Icon
      name={props.tab}
      size={20}
      color={props.activeTab === props.index ? COLORS.LIGHT_GREEN : 'gray'}
    />
  </TouchableOpacity>
);

Tab.propTypes = {
  tab: React.PropTypes.string,
  goToPage: React.PropTypes.func,
  index: React.PropTypes.number,
  activeTab: React.PropTypes.number
};

export default Tab;
