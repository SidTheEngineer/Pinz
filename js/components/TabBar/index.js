import React from 'react';
import { View, StyleSheet } from 'react-native';
import ScrollableTabView from 'react-native-scrollable-tab-view';
import Tab from './Tab';
import Map from '../Map';
import CategoryList from '../CategoryList';

const styles = StyleSheet.create({
  tabContainer: {
    height: 45,
    flexDirection: 'row',
  }
});

const CustomTabBar = props => (
  // Props here are coming from the ScrollableTabView component.
  <View style={styles.tabContainer}>
    {props.tabs.map((tab, index) => (
      <Tab
        {...props}
        index={index}
        tab={tab}
        key={tab}
      />
    ))}
  </View>
);

const TabBar = props => (
  <ScrollableTabView
    tabBarPosition="bottom"
    renderTabBar={() => <CustomTabBar />}
    locked
  >
    <Map tabLabel="map" {...props} />
    <CategoryList tabLabel="list" />
  </ScrollableTabView>
);

CustomTabBar.propTypes = {
  tabs: React.PropTypes.oneOfType([
    React.PropTypes.arrayOf(React.PropTypes.string),
    React.PropTypes.string
  ])
};

export default TabBar;
