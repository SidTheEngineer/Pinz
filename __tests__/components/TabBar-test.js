import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import TabBar from '../../js/components/TabBar';

// The TabBar is responsible for returning a MapView, so
// react-native-maps is needed, as well as initial props.

jest.mock('react-native-scrollable-tab-view', () => 'ScrollableTabView');
jest.mock('react-native-maps', () => 'MapView');
jest.mock('../../js/components/Map/EventMarker', () => 'EventMarker');

const props = {
  map: {
    modalVisibility: false,
    modalDetails: null,
    events: [
      {
        event_id: '12345',
        id: '12345',
        title: 'This is a test event',
        description: 'test test test',
        location: 'Location 123',
        starts: 'midnight',
        ends: 'midnight',
        category: 'Sports'
      }
    ]
  },
  mapActions: {
    fetchInitialEvents: jest.fn()
  }
};

describe('TabBar', () => {
  it('renders correctly', () => {
    const tabBar = renderer.create(
      <TabBar {...props} />
    ).toJSON();
    expect(tabBar).toMatchSnapshot();
  });
});
