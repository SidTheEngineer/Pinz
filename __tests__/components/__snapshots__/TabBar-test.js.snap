exports[`TabBar renders correctly 1`] = `
<ScrollableTabView
  locked={true}
  renderTabBar={[Function]}
  tabBarPosition="bottom">
  <View
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "flex-start",
        },
        Object {},
      ]
    }>
    <View
      style={
        Object {
          "backgroundColor": "rgb(51, 204, 94)",
          "height": 20,
        }
      } />
    <MapView
      loadingEnabled={true}
      moveOnMarkerPress={false}
      onRegionChange={[Function]}
      pitchEnabled={false}
      provider="google"
      region={
        Object {
          "latitude": 28.60166,
          "latitudeDelta": 0.0206,
          "longitude": -81.200788,
          "longitudeDelta": 0.014,
        }
      }
      rotateEnabled={false}
      showsCompass={false}
      style={
        Object {
          "flex": 5,
        }
      }>
      <EventMarker
        details={
          Object {
            "category": "Sports",
            "description": "test test test",
            "ends": "midnight",
            "event_id": "12345",
            "id": "12345",
            "location": "Location 123",
            "starts": "midnight",
            "title": "This is a test event",
          }
        }
        loadModal={[Function]} />
    </MapView>
    <Modal
      animationType="fade"
      onRequestClose={[Function]}
      transparent={true}
      visible={false}>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(20, 20, 20, 0.8)",
            "flex": 1,
          }
        }>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderColor": "rgb(150,150,150)",
              "borderRadius": 10,
              "borderStyle": "solid",
              "borderWidth": 0.5,
              "flex": 1,
              "flexDirection": "column",
              "marginBottom": 125,
              "marginLeft": 30,
              "marginRight": 30,
              "marginTop": 125,
            }
          }>
          <View
            style={
              Object {
                "alignItems": "center",
                "marginLeft": 25,
                "marginRight": 25,
              }
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignItems": "center",
                  "color": "rgb(100,100,100)",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "justifyContent": "center",
                  "marginTop": 25,
                }
              }>
              
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontWeight": "bold",
                  "marginTop": 30,
                }
              }>
              Starts: 
              
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontWeight": "bold",
                  "marginTop": 30,
                }
              }>
              Ends: 
              
            </Text>
            <ScrollView
              style={
                Object {
                  "marginBottom": 10,
                  "marginTop": 15,
                }
              }>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignItems": "center",
                    "color": "rgb(100,100,100)",
                    "fontSize": 16,
                    "justifyContent": "center",
                  }
                }>
                
              </Text>
            </ScrollView>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignItems": "center",
                    "color": "rgb(51, 204, 94)",
                    "fontSize": 18,
                    "justifyContent": "center",
                    "marginBottom": 20,
                  }
                }>
                Back
              </Text>
            </TouchableOpacity>
          </View>
        </View>
      </View>
    </Modal>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }>
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "overflow": "hidden",
        }
      }>
      <View
        onStartShouldSetResponderCapture={[Function]}
        pointerEvents="auto"
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            null,
          ]
        }>
        <ScrollView
          style={
            Object {
              "backgroundColor": "rgb(235,235,235)",
              "flex": 1,
              "marginTop": 64,
            }
          }>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "flexDirection": "row",
                "height": 75,
              }
            }>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "margin": 5,
                }
              }>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "rgb(100,100,100)",
                    "fontWeight": "bold",
                    "marginLeft": 15,
                  }
                }>
                Events
              </Text>
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "rgb(100,100,100)",
                      "fontSize": 20,
                    },
                    Object {
                      "marginRight": 15,
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }>
                
              </Text>
            </View>
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={undefined}
            style={
              Object {
                "flexDirection": "row",
                "height": 75,
              }
            }>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "margin": 5,
                }
              }>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "rgb(100,100,100)",
                    "fontWeight": "bold",
                    "marginLeft": 15,
                  }
                }>
                Garages
              </Text>
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "rgb(100,100,100)",
                      "fontSize": 20,
                    },
                    Object {
                      "marginRight": 15,
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }>
                
              </Text>
            </View>
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={undefined}
            style={
              Object {
                "flexDirection": "row",
                "height": 75,
              }
            }>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "margin": 5,
                }
              }>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "rgb(100,100,100)",
                    "fontWeight": "bold",
                    "marginLeft": 15,
                  }
                }>
                Food
              </Text>
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "rgb(100,100,100)",
                      "fontSize": 20,
                    },
                    Object {
                      "marginRight": 15,
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }>
                
              </Text>
            </View>
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={undefined}
            style={
              Object {
                "flexDirection": "row",
                "height": 75,
              }
            }>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "margin": 5,
                }
              }>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "rgb(100,100,100)",
                    "fontWeight": "bold",
                    "marginLeft": 15,
                  }
                }>
                Test
              </Text>
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "rgb(100,100,100)",
                      "fontSize": 20,
                    },
                    Object {
                      "marginRight": 15,
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }>
                
              </Text>
            </View>
          </TouchableOpacity>
        </ScrollView>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 64,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "rgb(79,232,122)",
            "height": 64,
            "justifyContent": "center",
          },
        ]
      }>
      <View
        pointerEvents="box-none"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "height": 44,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 20,
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignItems": "center",
              "color": "white",
              "flex": 1,
              "fontSize": 18,
              "fontWeight": "bold",
              "justifyContent": "center",
              "marginTop": 6,
            }
          }>
          Categories
        </Text>
      </View>
    </View>
  </View>
</ScrollableTabView>
`;
